provider/docker/builder_test.go

```diff
@@ -0,0 +1,69 @@
+package docker
+
+import (
+	docker "github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/container"
+	"github.com/docker/engine-api/types/network"
+	"github.com/docker/go-connections/nat"
+)
+
+func containerJSON(ops ...func(*docker.ContainerJSON)) docker.ContainerJSON {
+	c := &docker.ContainerJSON{
+		ContainerJSONBase: &docker.ContainerJSONBase{
+			Name:       "fake",
+			HostConfig: &container.HostConfig{},
+		},
+		Config: &container.Config{},
+		NetworkSettings: &docker.NetworkSettings{
+			NetworkSettingsBase: docker.NetworkSettingsBase{},
+		},
+	}
+
+	for _, op := range ops {
+		op(c)
+	}
+
+	return *c
+}
+
+func name(name string) func(*docker.ContainerJSON) {
+	return func(c *docker.ContainerJSON) {
+		c.ContainerJSONBase.Name = name
+	}
+}
+
+func networkMode(mode string) func(*docker.ContainerJSON) {
+	return func(c *docker.ContainerJSON) {
+		c.ContainerJSONBase.HostConfig.NetworkMode = container.NetworkMode(mode)
+	}
+}
+
+func labels(labels map[string]string) func(*docker.ContainerJSON) {
+	return func(c *docker.ContainerJSON) {
+		c.Config.Labels = labels
+	}
+}
+
+func ports(portMap nat.PortMap) func(*docker.ContainerJSON) {
+	return func(c *docker.ContainerJSON) {
+		c.NetworkSettings.NetworkSettingsBase.Ports = portMap
+	}
+}
+
+func withNetwork(name string, ops ...func(*network.EndpointSettings)) func(*docker.ContainerJSON) {
+	return func(c *docker.ContainerJSON) {
+		if c.NetworkSettings.Networks == nil {
+			c.NetworkSettings.Networks = map[string]*network.EndpointSettings{}
+		}
+		c.NetworkSettings.Networks[name] = &network.EndpointSettings{}
+		for _, op := range ops {
+			op(c.NetworkSettings.Networks[name])
+		}
+	}
+}
+
+func ipv4(ip string) func(*network.EndpointSettings) {
+	return func(s *network.EndpointSettings) {
+		s.IPAddress = ip
+	}
+}
```

Comment:
I'm on the fence whether the builder deserves its on set if tests. It doesn't seem strictly required but would be helpful to have them.